### ISSUE TITLE
How to improve the qualityof ReconstructMesh  about  small objects ？

### ISSUE THREAD

**Issue by Why384040146** [cdcseacave/openMVS#694]
hello, I have reconstructed point clouds of city buildings as figure1, but when I reconstrcutMesh about the pointcloud,  I found the small object which for example a slender object can't rebuild.
How can i improve the mesh quality about the slender object?
![pointcloud](https://user-images.githubusercontent.com/17980581/129999823-c2013cbc-f3bf-4823-83e3-56647deb6f0a.png)
![mesh](https://user-images.githubusercontent.com/17980581/129999838-fab9f6f4-e365-4c47-9acb-4f4cef04862f.png)

**Comment by cdcseacave** [comment]
indeed, that is a known and difficult problem, I am looking into ways to solve this, but for now you can try to get more detailed images of the objects that dissapear.

**Comment by Why384040146** [comment]
> indeed, that is a known and difficult problem, I am looking into ways to solve this, but for now you can try to get more detailed images of the objects that dissapear.
@cdcseacave 

I found that when I non-uniform sampling of the point cloud as that :   a slender object is dense and plane points is sparse .  the thin object can be constructed in mesh  as  above figure.   But  there is a problem the electric pole has Sharp triangle that  cant be removed appropriatly.   Should  I adjust the algorithm parameters, or How can I reduce the sharp triangle but retain the filled hole triangle ?
![mesh](https://user-images.githubusercontent.com/17980581/132300118-e1c9199e-9f18-401f-bd75-2d149fde71bf.png)

**Comment by cdcseacave** [comment]
you can try adjusting `--remove-spurious`

**Comment by Why384040146** [comment]
> you can try adjusting `--remove-spurious

Yes， I have been trying to the pare before, Indeed the  sharp triangle can be removed but the filled hole triangle  was removed together.   I cant distinguish the sharp triangle  generated by filling the hole.

**Comment by cdcseacave** [comment]
know significant progress on this yet,
however you can try increasing the image resolution and quality, and get closer to the object of interest

**Comment by BoyangHu1998** [comment]
> Hi @cdcseacave, has there been any progress in solving the slender object issue? I'm having trouble reconstructing thin poles using the ReconstructMesh function, and am wondering if there are any set of parameter values that can help in reconstruction those slender objects?
> 
> This is from an electrical pylon, which has many thin metal poles as supportive scafolding. I have disabled the cleaning portion of the reconstruction, to narrow down the issue to be mesh reconstruction. The dense pointcloud from from the DensifyPointCloud function shows each of the metal poles clearly.
> 
> ![Screenshot from 2024-03-12 17-43-28](https://private-user-images.githubusercontent.com/57743211/312025930-27ac7db8-ae62-4a09-a0ec-7cb506945825.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ0OTgxMTgsIm5iZiI6MTczNDQ5NzgxOCwicGF0aCI6Ii81Nzc0MzIxMS8zMTIwMjU5MzAtMjdhYzdkYjgtYWU2Mi00YTA5LWEwZWMtN2NiNTA2OTQ1ODI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE4VDA0NTY1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJjNzdhYWZjMjcwODZkZTdjMjlkNzg3Y2MxNTU2YTdhYzIzYmViMDE4OWJkYTE4YmNmMmRkOTk3NjU2NDhmNjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ZVZu_xm0CRUe9gtMS8jPfYQNK6pX2eptWdMsPX7Y0f8)

Hi @cdcseacave, I wanted to follow up on the slender object reconstruction issue. I'm still encountering similar challenges reconstructing thin poles, such as those on an electrical pylon, using the ReconstructMesh function. Are there any updates, parameter recommendations, or alternative approaches to solve this issue?

Any insights or progress would be greatly appreciated!

Thanks!

**Comment by cdcseacave** [comment]
In order to reconstruct thin surfaces you need perfect poses, images, and high resolution.
Once you have all the above, you can also try using '-d 1' or smaller, or even zero.

**Comment by cdcseacave** [comment]
Closing as stale. The codebase has been substantially updated on the \develop\ branch. Please reopen with reproduction steps if the issue persists on the latest code.
